### PR TITLE
Use `set` instead of `True`-only `dict`

### PR DESCRIPTION
**Pull request by Avasam**
## Summary of changes
Use `set` instead of `True`-only `dict`

Note this is likely a breaking changes for some downstream users of
- `pkg_resources.ResourceManager.cached_files`
- `setuptools.package_index.PackageIndex.package_pages`'s values
- `setuptools.package_index.PackageIndex.scanned_urls`
- `setuptools.package_index.PackageIndex.fetched_urls`

if they access of modify its content.

### Pull Request Checklist
- [ ] Changes have tests (existing tests should pass + static type-checking)
- [x] News fragment added in [`newsfragments/`].
  _(See [documentation][PR docs] for details)_

[`newsfragments/`]: https://github.com/pypa/setuptools/tree/master/newsfragments
[PR docs]:
https://setuptools.pypa.io/en/latest/development/developer-guide.html#making-a-pull-request

### CODE REVIEW

File: setuptools/package_index.py [hunk 1]
```diff
@@ -310,9 +311,9 @@ def __init__(
     ):
         super().__init__(*args, **kw)
         self.index_url = index_url + "/"[: not index_url.endswith('/')]
-        self.scanned_urls = {}
-        self.fetched_urls = {}
-        self.package_pages = {}
+        self.scanned_urls: Set[str] = set()
+        self.fetched_urls: Set[str] = set()
+        self.package_pages: Dict[str, Set[str]] = {}
```
[the remark this answers is not in the document]
Comment (Avasam) [reply setuptools/package_index.py:316]: `{}` is a literal shorthand for an empty `dict`, not an empty `set`.

File: setuptools/tests/test_easy_install.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def test_no_find_links(self):
         cmd.install_dir = os.path.join(tempfile.mkdtemp(), 'ok')
         cmd.args = ['ok']
         cmd.ensure_finalized()
-        assert cmd.package_index.scanned_urls == {}
+        assert len(cmd.package_index.scanned_urls) == 0
```
[the remark this answers is not in the document]
Comment (Avasam) [reply setuptools/tests/test_easy_install.py:93]: I do prefer being being explicit in a test. In fact I could've left this as
```suggestion
        assert cmd.package_index.scanned_urls == set()
```
As it also validates the type (either that or 1 assert for type check and 1 assert for content check).

File: pkg_resources/__init__.py [hunk 3]
```diff
@@ -1239,7 +1239,7 @@ def get_cache_path(self, archive_name, names=()):
 
         self._warn_unsafe_extraction_path(extract_path)
 
-        self.cached_files[target_path] = 1
+        self.cached_files[target_path] =
```
[the remark this answers is not in the document]
Comment (Avasam) [reply pkg_resources/__init__.py]: There's a bit more to be changed, but whoops! Thanks. I rush pushed that one before the weekend.

File: setuptools/package_index.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DimitriPapadopoulos) [reply setuptools/package_index.py:316]: Indeed, got mixed up by https://github.com/fsspec/filesystem_spec/pull/1602#discussion_r1606882866

**Comment by abravalheri** [comment]
Thank you very much for the continuous contributions, @Avasam 

> Note this is likely a breaking changes for some downstream

Can we live without this one? Maybe if it is a breaking change we can stop back?

**Comment by Avasam** [comment]
> Maybe if it is a breaking change we can step back?

Since v70 just got published, I'll make a version of this PR that doesn't touch public names (let me know if any of the 4 public names listed above were not meant as public api anyway). So this can be considered for v71 instead.

> Will it bring any other improvement?

I don't think so, it's just a datatype improvement.

**Comment by abravalheri** [comment]
Thank you @Avasam, I merged #4381. I think we can close this PR for now.

Since `pkg_resources` is considered deprecated, it should be fine to keep True-only dicts and avoid breaking the API for the rest of the life-cycle of the package.

**Comment by Avasam** [comment]
> Since `pkg_resources` is considered deprecated, it should be fine to keep True-only dicts and avoid breaking the API for the rest of the life-cycle of the package.

Sure for `pkg_resources`, what about the setuptools change? I don't mind if this takes a while to be merged waiting for the next major version bump. It's tangential to anything else I was doing.

**Comment by abravalheri** [comment]
Should we drop this to avoid changing public names?

**Comment by Avasam** [comment]
> Should we drop this to avoid changing public names?

Given 
> `PackageIndex` is kind of associated with `easy_install` so is kind of deprecated, or at least "soft-deprecated" as well isn't it (at least in my eyes...)

ok yeah

**Pull request closed without merging**